content/blog/2019-01-07-we-mastered-the-art-of-programming.md

```diff
@@ -0,0 +1,63 @@
+
+Maybe if you took a little more time to think of how future you would feel using it things would have been different. Maybe you would have cleaned the code. Maybe you would have explicated the [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) (why divide by 8.2?). Maybe some side effects would have been commented. Well this is not important for now, because the code *works*.
+
+But this did not apply to me. I could jump back in my code, read the witty comment I wrote and remember what dimension of the matrix `X` were represented by variable `i`, `j` and `k`. I make functions, so I [don't repeat myself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). I had a bunch of smell tests that let me know if I break something. No really, I mastered the art of programming.
+
+## Until one day.
+
+I was working in a [Tech Startup](https://www.xpertsea.com/) and my team was working on a shrimp detection model. We were a few guys with different backgrounds. Everyone were software engineers except for myself; I was an actuary holding the most statistical knowledge. We were pitted against one another to figure how to solve this shrimp detection problem. Plot twist. I was not the one achieving — and for reasons I was not even aware of.
+
+It started pretty well. I quickly build a script that let me grab the data from an unstructured source and convert it to a structured format. I would then run some diagnosis to find any outliers or collinearity between features. Once the data was cleaned, another script generated a regression model using a step-wise algorithm. The model would generate some predictions and save them into a file. The last script was doing some graph and some basic analysis. Pretty standard data science duty. The report was done deal and I could quickly deliver some great results to the team.
+
+Then we had to evaluate the evolution of the model. This analysis became a weekly affair, then a daily one. Every day, the analysis got more complicated and the dataset got bigger and weirder. We cross-sectioned the data for certain shrimp providers, shrimp age, average size of shrimp, country, quality of water — name it. We changed the images processing method, we added countless custom features, we used various metrics, we corrected the dataset, etc. This is where I hit the wall. The software engineer guy was pumping his analysis a few hours before me, every day.
+
+## Why isn't this working ?
+
+This called for my problem solving skills. How can I achieve faster production? By better understanding what he was doing.
```

Comment:
This called : pas certain c'est quoi le is, surtout que tu change de section. this issue, fuckup, setback ?